esi/lib/networks.py

```diff
@@ -112,3 +112,44 @@ def get_networks_from_port(connection, port, networks_dict={}, floating_ips_dict
         floating_network = connection.network.get_network(floating_network_id)
 
     return parent_network, trunk_networks, trunk_ports, floating_network
+
+
+def create_port(connection, network, name=None):
+    """
+    Creates a port on the specified network using the network object.
+
+    :param connection: An OpenStack connection object
+    :param network: The network object where the port should be created
+    :param name: Optional name for the port. If not provided, a default name is generated.
+
+    :return: The created port object
+    """
+    if not name:
+        name = 'esi-port-{0}'.format(network.name)
```

Comment:
Can you name the port the same way we do here? https://github.com/CCI-MOC/esisdk/blob/main/esi/lib/nodes.py#L212

It actually probably makes sense to create a new `port_name` function, and just call it both here and in the place that I linked.